scripts/console/2025.0/001_retire_vlans.py

```diff
@@ -0,0 +1,147 @@
+        ):
+            tag = evc[uni]["tag"]["value"]
+            if isinstance(tag, int):
+                in_use_tags[intf_id].append((tag, evc_id))
+            elif isinstance(tag, list):
+                for tag_item in tag:
+                    if isinstance(tag_item, int):
+                        in_use_tags[intf_id].append((tag_item, evc_id))
+                    elif isinstance(tag_item, list) and len(tag_item) == 1:
+                        in_use_tags[intf_id].append((tag_item[0], evc_id))
+                    elif isinstance(tag_item, list) and len(tag_item) == 2:
+                        for val in range(tag_item[0], tag_item[1]+1):
+                            in_use_tags[intf_id].append((val, evc_id))
+
+switch_rm_flows = {}
+
+evc_disable_set = set()
+
+dry_run_key = "WILL" if not DRY_RUN else "WOULD"
+
+print("Checking EVCs for vlan usage...")
+
+# Find all evcs to temporarily disable
+for intf_id in in_use_tags:
+    for tag, evc_id in in_use_tags[intf_id]:
+        if evc_id in evc_disable_set:
+            continue
+        intersect = range_helpers.range_intersection([[tag, tag]], RETIRED_VLANS)
+        intersect = list(intersect)
+        if intersect:
+            print(
+                f"EVC {evc_id} is using s_vlan {intersect} which is pending retirement, {dry_run_key} temporarily disable it..."
+            )
+            evc_disable_set.add(evc_id)
+
+# Disable EVCs
+if not DRY_RUN and evc_disable_set:
+    print("Disabling EVCs...")
+    for evc_id in evc_disable_set:
+            disable_evc(evc_id)
+
+# Retire the given vlan ranges
+
+if not DRY_RUN:
+    print("Clearing vlan from tag_ranges and available_tags")
+else:
+    print("Checking interfaces for vlan usage...")
+
+for dpid in list(controller.switches.keys()):
+    switch = controller.get_switch_by_dpid(dpid)
+    for intf_id, intf in switch.interfaces.copy().items():
+        with intf._tag_lock:
+            used_tags = range_helpers.range_difference(
+                intf.tag_ranges['vlan'], intf.available_tags['vlan']
+            )
+            new_range = range_helpers.range_difference(
+                intf.tag_ranges['vlan'], RETIRED_VLANS
+            )
+            missing = range_helpers.range_difference(
+                used_tags, new_range
+            )
+            if missing:
+                if not DRY_RUN:
+                    print(
+                        f"WARNING: Interface {dpid} {intf_id} still has the vlans {missing} in use."
+                    )
+                else:
+                    print(
+                        f"Interface {dpid} {intf_id} has the vlans {missing} in use."
+                    )
+                continue
+            if not DRY_RUN:
+                new_available_tags = range_helpers.range_difference(
+                    new_range, used_tags
+                )
+                intf.available_tags['vlan'] = new_available_tags
+                intf.tag_ranges['vlan'] = new_range
```

Comment:
A new tag range value needs to be written in the DB with [topology.handle_on_interface_tags(interface)](https://github.com/kytos-ng/topology/blob/master/main.py#L895). 

Let's also only do it when the `new_range` range_difference is different from `intf.tag_ranges['vlan']`, so when it intersected with `RETIRED_VLANS`